gradio/blocks.py

```diff
@@ -634,10 +639,12 @@ def launch(
         utils.launch_counter()
 
         # If running in a colab or not able to access localhost,
-        # automatically create a shareable link.
-        is_colab = utils.colab_check()
+        # a shareable link must be created.
         if is_colab or (_frontend and not networking.url_ok(self.local_url)):
-            share = True
+            if not self.share:
```

Comment:
We should only raise this exception if a user explicitly sets `share=False` in colab. It looks like this exception will be raised even if they call `.launch()` without passing in an explicit value for `None`